### ISSUE TITLE
Asus Tinkerboard install failing

### ISSUE THREAD

**Issue by Magno99900** [MichaIng/DietPi#2977]
------------------------------------------------------------------
- DietPi has encounted an error, and, is unable to continue        -
- Please create a ticket: https://github.com/Fourdee/DietPi/issues -
- Copy and paste the BLUE lines below, into the ticket             -
--------------------------------------------------------------------


#### Details:
- Date           | Wed 10 Jul 05:36:14 BST 2019
- Bug report     | N/A
- DietPi version | v6.18.14 (Fourdee/master)
- Img creator    | DietPi Core Team
- Pre-image      | ARMbian
- SBC device     | Asus Tinker Board (armv7l) (index=52)
- Kernel version | #270 SMP PREEMPT Wed Sep 19 11:40:40 CEST 2018
- Distro         | stretch (index=4)
- Command        | G_AGUG
- Exit code      | 100
- Software title | DietPi-Update

#### Steps to reproduce:
<!-- Explain how to reproduce the issue -->
1. trying to install latest version of dietpi from download page onto known working asus tinkerboard.
2. ...

#### Expected behaviour:
<!-- What SHOULD be happening? -->
- ...

#### Actual behaviour:
<!-- What IS happening? -->
- ...

#### Extra details:
<!-- Please post any extra details that might help solve the issue -->
- Tried multiple SD cards and tried re downloading the file

#### Additional logs:
```
Log file contents:
Setting up apt-utils (1.4.9) ...
Setting up tzdata (2019a-0+deb9u1) ...

Current default time zone: 'Europe/London'
Local time is now:      Wed Jul 10 05:35:37 BST 2019.
Universal Time is now:  Wed Jul 10 04:35:37 UTC 2019.
Run 'dpkg-reconfigure tzdata' if you wish to change it.

Setting up systemd-sysv (232-25+deb9u11) ...
Setting up linux-u-boot-tinkerboard-next (5.90) ...
Setting up unzip (6.0-21+deb9u1) ...
Setting up armbian-firmware (5.90) ...
Setting up libssl1.0.2:armhf (1.0.2s-1~deb9u1) ...
Setting up libssh2-1:armhf (1.7.0-1+deb9u1) ...
Processing triggers for libc-bin (2.24-11+deb9u4) ...
Setting up udev (232-25+deb9u11) ...
addgroup: The group `input' already exists as a system group. Exiting.
update-initramfs: deferring update (trigger activated)
Setting up libssl1.1:armhf (1.1.0k-1~deb9u1) ...
Setting up libc-l10n (2.24-11+deb9u4) ...
Processing triggers for systemd (232-25+deb9u11) ...
Setting up openssl (1.1.0k-1~deb9u1) ...
Setting up wget (1.18-5+deb9u3) ...
Setting up htop (2.1.2-3) ...
Setting up gnupg-agent (2.1.18-8~deb9u4) ...
Setting up dirmngr (2.1.18-8~deb9u4) ...
Setting up armbian-tools-stretch (5.75) ...
Setting up linux-dtb-next-rockchip (5.90) ...
Setting up locales (2.24-11+deb9u4) ...
Generating locales (this might take a while)...
  en_GB.UTF-8... done
Generation complete.
Setting up libdbus-1-3:armhf (1.10.28-0+deb9u1) ...
Setting up libcurl3:armhf (7.52.1-5+deb9u9) ...
Setting up libcurl3-gnutls:armhf (7.52.1-5+deb9u9) ...
Setting up gnupg (2.1.18-8~deb9u4) ...
Setting up libdns-export162 (1:9.10.3.dfsg.P4-12.3+deb9u5) ...
Setting up wpasupplicant (2:2.4-1+deb9u4) ...
Setting up apt-transport-https (1.4.9) ...
Setting up curl (7.52.1-5+deb9u9) ...
Processing triggers for initramfs-tools (0.130) ...
update-initramfs: Generating /boot/initrd.img-4.19.57-rockchip
/etc/initramfs/post-update.d//99-uboot: 3: .: Can't open /etc/armbian-release
run-parts: /etc/initramfs/post-update.d//99-uboot exited with return code 2
dpkg: error processing package initramfs-tools (--configure):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for libc-bin (2.24-11+deb9u4) ...
Errors were encountered while processing:
 initramfs-tools
E: Sub-process /usr/bin/dpkg returned an error code (1)
```


--------------------------------------------------------------------

[FAILED] DietPi-Update | Unable to continue, DietPi-Update will now terminate.
 ────────────────────────────────────────────────
 DietPi | Update available
 ────────────────────────────────────────────────
 v6.18.14 | Asus Tinker Board (armv7l)
 ────────────────────────────────────────────────
 eth0 | 192.168.1.53
 ────────────────────────────────────────────────
[FAILED] DietPi-Software |
 Installation aborted by user.
 Installation must be completed prior to using DietPi.
 Please run dietpi-software to restart the installation.

root@DietPi:~# dietpi-software
[  OK  ] DietPi-Software | Root access verified.
[  OK  ] DietPi-Software | RootFS R/W access verified.

[  OK  ] DietPi-Software | Initialized database
[  OK  ] DietPi-Software | Reading database
[  OK  ] DietPi-Software | Connection test: https://deb.debian.org/debian/
[  OK  ] NTPD: time sync | Completed
[  OK  ] DietPi-Config | File exists | /usr/share/i18n/SUPPORTED
[ SUB2 ] DietPi-Set_software > locale (en_US.UTF-8)
[  OK  ] DietPi-Set_software | dpkg-reconfigure -f noninteractive locales
[  OK  ] DietPi-Set_software | Setting in /DietPi/dietpi.txt adjusted: AUTO_SETUP_LOCALE=en_US.UTF-8
[  OK  ] locale en_US.UTF-8 | Completed

Current default time zone: 'America/Los_Angeles'
Local time is now:      Tue Jul  9 21:37:22 PDT 2019.
Universal Time is now:  Wed Jul 10 04:37:22 UTC 2019.
[  OK  ] DietPi-Config | Connection test: https://google.com
[ SUB2 ] DietPi-Set_Hardware > preferipv4 (disable)
[  OK  ] DietPi-Set_Hardware | Setting in /DietPi/dietpi.txt adjusted: CONFIG_PREFER_IPV4=0
[  OK  ] preferipv4 disable | Completed
[ SUB2 ] DietPi-Set_software > apt-mirror (https://deb.debian.org/debian/)
[  OK  ] DietPi-Set_software | Desired setting in /DietPi/dietpi.txt was already set: CONFIG_APT_DEBIAN_MIRROR=https://deb.debian.org/debian/
[  OK  ] apt-mirror https://deb.debian.org/debian/ | Completed
[  OK  ] DietPi-Config | APT update, please wait...
Ign:2 https://cdn-aws.deb.debian.org/debian stretch InRelease
Hit:3 https://cdn-aws.deb.debian.org/debian stretch-updates InRelease
Hit:4 https://cdn-aws.deb.debian.org/debian-security stretch/updates InRelease
Hit:5 https://cdn-aws.deb.debian.org/debian stretch-backports InRelease
Hit:1 https://apt.armbian.com stretch InRelease
Hit:6 https://cdn-aws.deb.debian.org/debian stretch Release
Reading package lists...

[  OK  ] DietPi-Config | G_AGUP
[  OK  ] DietPi-Software | Free space check: path=/ | available=27842 MB | required=500 MB
[  OK  ] DietPi-Software | DietPi-Userdata validation: /mnt/dietpi_userdata
[  OK  ] DietPi-Software | Connection test: https://deb.debian.org/debian/
[  OK  ] NTPD: time sync | Completed
[ SUB1 ] DietPi-Services > unmask
[  OK  ] DietPi-Services | unmask all: cron
[ SUB1 ] DietPi-Services > stop
[  OK  ] DietPi-Services | stop : cron

 DietPi-Software
─────────────────────────────────────────────────────
 Mode: Update & upgrade APT

[  OK  ] DietPi-Software | APT update, please wait...
Ign:1 https://cdn-aws.deb.debian.org/debian stretch InRelease
Hit:2 https://cdn-aws.deb.debian.org/debian stretch-updates InRelease
Hit:3 https://cdn-aws.deb.debian.org/debian-security stretch/updates InRelease
Hit:4 https://cdn-aws.deb.debian.org/debian stretch-backports InRelease
Hit:6 https://cdn-aws.deb.debian.org/debian stretch Release
Hit:5 https://apt.armbian.com stretch InRelease
Reading package lists...

[  OK  ] DietPi-Software | G_AGUP
[ INFO ] DietPi-Software | Running apt simulation to check for errors, please wait...
[  OK  ] DietPi-Software | APT installation for: bash-doc -s, please wait...
Inst bash-doc (4.4-5 Debian:9.9/oldstable [all])
Conf bash-doc (4.4-5 Debian:9.9/oldstable [all])
Conf initramfs-tools (0.130 Debian:9.9/oldstable [all])

[  OK  ] DietPi-Software | G_AGI: bash-doc -s
[  OK  ] DietPi-Software | APT upgrade, please wait...
Reading package lists...
                        Building dependency tree...
                                                   Reading state information...
                                                                               Calculating upgrade...
                                                                                                     The following packages have been kept back:
                                                                                                                                                  linux-stretch-root-next-tinkerboard
                                                                                                                                                                                     0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
                                                                                                                                                                                                                                                   1 not fully installed or removed.
     After this operation, 0 B of additional disk space will be used.
                                                                     Setting up initramfs-tools (0.130) ...
update-initramfs: deferring update (trigger activated)
Processing triggers for initramfs-tools (0.130) ...
update-initramfs: Generating /boot/initrd.img-4.19.57-rockchip
/etc/initramfs/post-update.d//99-uboot: 3: .: Can't open /etc/armbian-release
run-parts: /etc/initramfs/post-update.d//99-uboot exited with return code 2
dpkg: error processing package initramfs-tools (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 initramfs-tools
E: Sub-process /usr/bin/dpkg returned an error code (1)

[FAILED] DietPi-Software | G_AGUG
[  OK  ] DietPi-Software | APT upgrade, please wait...
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages have been kept back:
  linux-stretch-root-next-tinkerboard
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
                                                              1 not fully installed or removed.
                                                                                               After this operation, 0 B of additional disk space will be used.
                                                                                                                                                               Setting up initramfs-tools (0.130) ...
update-initramfs: deferring update (trigger activated)
Processing triggers for initramfs-tools (0.130) ...
update-initramfs: Generating /boot/initrd.img-4.19.57-rockchip
/etc/initramfs/post-update.d//99-uboot: 3: .: Can't open /etc/armbian-release
run-parts: /etc/initramfs/post-update.d//99-uboot exited with return code 2
dpkg: error processing package initramfs-tools (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 initramfs-tools
E: Sub-process /usr/bin/dpkg returned an error code (1)

[FAILED] DietPi-Software | G_AGUG


--------------------------------------------------------------------
- DietPi has encounted an error, and, is unable to continue        -
- Please create a ticket: https://github.com/Fourdee/DietPi/issues -
- Copy and paste the BLUE lines below, into the ticket             -
--------------------------------------------------------------------


#### Details:
- Date           | Tue  9 Jul 21:39:26 PDT 2019
- Bug report     | N/A
- DietPi version | v6.18.14 (Fourdee/master)
- Img creator    | DietPi Core Team
- Pre-image      | ARMbian
- SBC device     | Asus Tinker Board (armv7l) (index=52)
- Kernel version | #270 SMP PREEMPT Wed Sep 19 11:40:40 CEST 2018
- Distro         | stretch (index=4)
- Command        | G_AGUG
- Exit code      | 100
- Software title | DietPi-Software

#### Steps to reproduce:
<!-- Explain how to reproduce the issue -->
1. ...
2. ...

#### Expected behaviour:
<!-- What SHOULD be happening? -->
- ...

#### Actual behaviour:
<!-- What IS happening? -->
- ...

#### Extra details:
<!-- Please post any extra details that might help solve the issue -->
- ...

#### Additional logs:
```
Log file contents:
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages have been kept back:
  linux-stretch-root-next-tinkerboard
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Setting up initramfs-tools (0.130) ...
update-initramfs: deferring update (trigger activated)
Processing triggers for initramfs-tools (0.130) ...
update-initramfs: Generating /boot/initrd.img-4.19.57-rockchip
/etc/initramfs/post-update.d//99-uboot: 3: .: Can't open /etc/armbian-release
run-parts: /etc/initramfs/post-update.d//99-uboot exited with return code 2
dpkg: error processing package initramfs-tools (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 initramfs-tools
E: Sub-process /usr/bin/dpkg returned an error code (1)
```


--------------------------------------------------------------------

[FAILED] DietPi-Software | Unable to continue, DietPi-Software will now terminate.

root@DietPi:~#

**Comment by MichaIng** [comment]
@Magno99900 
Many thanks for your report.

Please do the following:
```
cat << _EOF_ > /etc/armbian-release
# PLEASE DO NOT EDIT THIS FILE
BOARD=tinkerboard
BOARD_NAME="Tinkerboard"
BOARDFAMILY=rockchip
BUILD_REPOSITORY_URL=https://github.com/armbian/rkbin
BUILD_REPOSITORY_COMMIT=51d55a4
VERSION=5.90
LINUXFAMILY=rockchip
BRANCH=default
ARCH=arm
IMAGE_TYPE=stable
BOARD_TYPE=conf
INITRD_ARCH=arm
KERNEL_IMAGE_TYPE=zImage
_EOF_
apt-get -y upgrade
```

We need to recreate the ATB image.

**Comment by Fourdee** [comment]
@MichaIng 

Buster image uploaded:
https://dietpi.com/downloads/images/DietPi_ASUSTB-ARMv7-Buster.7z

I've ran a basic install test, seems fine. GPU drivers will probably not transition over from Stretch, needs testing when I/comminuty get time.

Done:
- :u6307: Replace dietpi.com with new buster image link
---

REF:
![image](https://user-images.githubusercontent.com/7077026/62424541-c87ea180-b6c7-11e9-8b66-82d3b51ffd0d.png)

**Comment by Fourdee** [comment]
- :u5408: GPU accel
- Breaks all X11/desktop env's.
- :u5408:  ```rm /etc/X11/xorg.conf```
- Packages appear to break APT. 
- :u6307: We need to remove GPU driver install from our code. Done with Dev branch for now.

**Comment by MichaIng** [comment]
@Fourdee 
Many thanks for creating this. Indeed our GPU driver packages are Stretch versions. I made and totally forgot some updated archive, however it provides Stretch versions as well: https://dietpi.com/downloads/binaries/asus/testing/

There are GPU drivers for t76x available in Debian Buster repo: [libegl1](https://packages.debian.org/buster/libegl1)
Not 100% sure which package is pulled by default by [libegl1-mesa](https://packages.debian.org/buster/libegl1-mesa), which we install with X.org, but I guess best would be: [mali-t76x-fbdev-driver](https://packages.debian.org/buster/mali-t76x-fbdev-driver)
Simple `xorg.conf` to start with:
```
cat << _EOF_ > /etc/X11/xorg.conf
Section "Device"
	Identifier	"Mali-Fbdev"
	Driver		"fbdev"
	Option		"fbdev"		"/dev/fb0"
EndSection
_EOF_
```

Sadly https://tinkerboarding.co.uk is down to get resources from.

Found an TinkerOS fork on Debian Buster basis that has Kodi pre-installed: https://sourceforge.net/projects/tinkerboard/
Will check for working drivers/configs contained. **EDIT: A 3rd party repo with GL drivers is contained: http://repo.linaro.org/ubuntu/linaro-overlay**
__________
I re-enabled previous GPU support on Stretch only: https://github.com/MichaIng/DietPi/pull/3035/commits/afd894be176e90d58a80bd5ffb7ef854e17da896
- Moved our updated drivers archive in place. The contained package versions are newer then the ones on Debian Stretch, thus no setting on hold required and most likely Stretch will never receive updates again. Also the mali lib symlink creation is not required by this and as bonus some additional media features are included. Has been proven to otherwise work as before: https://github.com/MichaIng/DietPi/issues/2296#issuecomment-500128433

**Comment by MichaIng** [comment]
I mark this as closed. GPU driver tests will be done in a separate issue.
